bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/StringListType.java

```diff
@@ -100,4 +101,30 @@ public static StringListType valueOf(String value) {
         return new StringListType(value);
     }
 
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((typeDetails == null) ? 0 : typeDetails.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(@Nullable Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        StringListType other = (StringListType) obj;
+        if (!typeDetails.equals(other.typeDetails)) {
+            return false;
+        }
+        return true;
```

Comment:
We can return the result of the comparison directly:

```suggestion
        return typeDetails.equals(other.typeDetails);
```